packages/headless/src/controllers/triggers/headless-query-trigger.ts

```diff
@@ -0,0 +1,88 @@
+import {SearchEngine} from '../../app/search-engine/search-engine';
+import {ConfigurationSection, TriggerSection} from '../../state/state-sections';
+import {configuration, triggers} from '../../app/reducers';
+import {buildController, Controller} from '../controller/headless-controller';
+import {loadReducerError} from '../../utils/errors';
+import {executeSearch} from '../../features/search/search-actions';
+import {
+  updateQuery,
+  UpdateQueryActionCreatorPayload,
+} from '../../features/query/query-actions';
+import {logQueryTrigger} from '../../features/triggers/trigger-analytics-actions';
+import {clearQueryTrigger} from '../../features/triggers/trigger-actions';
+
+/**
+ * The `QueryTrigger` controller handles query triggers.
+ */
+export interface QueryTrigger extends Controller {
+  /**
+   * the state of the `QueryTrigger` controller.
+   */
+  state: QueryTriggerState;
+}
+
+/**
+ * A scoped and simplified part of the headless state that is relevant to the `QueryTrigger` controller.
+ */
+export interface QueryTriggerState {
+  /**
+   * The new query to perform a search with after receiving a query trigger.
+   */
+  newQuery: string | null;
+
+  /**
+   * The query used to perform the search that received a query trigger in its response.
+   */
+  prevQuery: string | undefined;
+}
+
+/**
+ * Creates a `QueryTrigger` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @returns A `QueryTrigger` controller instance.
+ * */
+export function buildQueryTrigger(engine: SearchEngine): QueryTrigger {
+  if (!loadQueryTriggerReducers(engine)) {
+    throw loadReducerError;
+  }
+
+  const controller = buildController(engine);
+  const {dispatch} = engine;
+
+  const getState = () => engine.state;
+
+  return {
+    ...controller,
+
+    subscribe(listener: () => void) {
+      const strictListener = () => {
+        if (getState().triggers.query) {
```

Comment:
this will pass when the query is `''`, but won't pass if it's set back to `null`

Reply:
This doesnt pass if query is `''`. I tested it to make sure.

Reply:
the empty string is treated the same as null or undefined.

Reply:
You're right it's falsy https://developer.mozilla.org/en-US/docs/Glossary/Falsy